MonstieWash/Assets/Scripts/Consumables/ConsumablesManager.cs

```diff
@@ -0,0 +1,300 @@
+using JetBrains.Annotations;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System.Xml;
+
+public class ConsumablesManager : MonoBehaviour
+{
+
+    [Tooltip("How far apart the consumables are in the UI")] [SerializeField] private int uiBorderDistance; //Distance between objects in the Ui;
+    [Tooltip("How quickly Consumables move towards their open position")] [SerializeField] private float uiMoveSpeed; //How fast Ui Elements close and open;
+    [Tooltip("Sprite to represent open bag.")] [SerializeField] private Sprite openSprite; //Sprite used to represent open bag.
+    [Tooltip("Sprite to represent closed bag.")] [SerializeField] private Sprite closedSprite; //Sprite used to represent closed bag.
+    [SerializeField] private List<UIConsumable> activeUiElements; //UI Elements already generated in the scene;
+    [Tooltip("Gameobject new Ui consumable items will be parented under")] [SerializeField] private GameObject imageHolder; //Empty Gameobject in Ui to hold images;
+    [Tooltip("A template obj requiring, image, UIconsumableScript, button linked to the scripts onclick")] [SerializeField] private GameObject refObject; //blank object for instantiate to reference; Needs a button, UIConsumableScript, Image.
+    [Tooltip("Layer that checks player mouse is over monster collider")] [SerializeField] public LayerMask monsterLayer; //Layer of monster hitbox;
+
+    [Header("All consumables go here")]
+    [Tooltip("List all starting consumables")] [SerializeField] private List<Consumable> consumableData;
+
+    private Image managerimage; //Reference to the image object for this manager.
+    private Dictionary<string, int> storedDict; //string is name of consumable type, int is stored amount.
```

Comment:
Nearly forgot but these should be named with the appropriate prefix given they are pure member variables.
```suggestion
    private Image m_managerimage; //Reference to the image object for this manager.
    private Dictionary<string, int> m_storedDict; //string is name of consumable type, int is stored amount.
```